CMakeLists.txt

```diff
@@ -191,6 +193,8 @@ option_with_default(SPHINX_THEME "Theme for Sphinx documentation and extensions"
 option_with_default(SPHINXMAN_STRICT "Turn warnings into errors for docs target sphinxman" OFF)
 option_with_default(BUILD_Libint2_GENERATOR "If prebuilt not found, don't build Libint2 from source tarball but instead build and run generator (several hours for production library). Edit angular momentum and integrals classes directly in `external/upstream/libint2/CMakeLists.txt` (requires Ninja)." OFF)
 option_with_default(psi4_SHGAUSS_ORDERING "Ordering for shells of solid harmonic Gaussians: standard -- standard ordering (-l, -l+1 ... l) -or- gaussian -- the Gaussian ordering (0, 1, -1, 2, -2, ... l, -l). Changing this is experimental, expert only, and not a future plan. Internal calculations should be correct and fchk files should be correct, but other exports like Molden and QCSchema.Wavefunction won't be." gaussian)
+option_with_default(gauxc_ENABLE_GPU "Enable GPU support for GauXC" OFF)
+option_with_default(gauxc_ENABLE_MAGMA "Enable MAGMA support for GauXC. Required GPU-enabled GauXC" OFF)
```

Comment:
Are these if-gauxc-built-with-gpu/magma-then-enable-extra-code or demand-to-use-a-gpu/magma-enabled-build-of-gauxc-with-psi4 options? If the latter, should we request upstream components for `find_package(gauxc COMPONENTS gpu)` etc.? (I'm sorry to pile on cmake changes -- hopefully all-at-once is less, not more, annoying.)

Reply:
Yes, if you ask e.g. `GAUXC_ENABLE_CUDA`, you will get a configure failure if CUDA support is not available - so in that sense it's a demand, not a request.

Re components: I think we may view components slightly differently - I tend to view them as a linkable target within a package (e.g. in Libint2 we have `Libint2::libint2_Eigen Libint2::libint2 Libint2::libint2_cxx`) - there's no such distinction possible in GauXC - it's a single component with enabled features. Feature checks can be made via the exported variables that you and @davpoolechem added (e.g. `GAUXC_HAS_DEVICE`). My suggestion would be that if you set `gauxc_ENABLE_GPU`, you should check `GAUXC_HAS_DEVICE` is consistent in the discovered target. I suppose I could make this a component as well, but then we have multiple potential sources of truth (e.g. variables vs components), and unless we add robust UTs to check that these are consistent, this might be asking for trouble down the line.

Reply:
The `gauxc_ENABLE_PACKAGE` commands Psi4-side are a demand, like in GauXC. If building an internal GauXC instance, the arguments are passed in to the internal GauXC build, where it acts as a demand there. If using an external GauXC instance, the external GauXC instance must have the capabilities requested from the `gauxc_ENABLE_PACKAGE` keyword, or else the Psi4 compilation will fail. 

Like @wavefunction91 suggested, the Psi4/GauXC interface build uses GauXC's exported properties to make sure that the build capabilities line up with that requested Psi4-side. I don't see a big problem just using exported properties to validate build features, except for the fact that it could get a bit unwieldy as more GauXC build features are supported (e.g., I'm looking into the possibility of Cutlass support for the Psi4/GauXC interface).